src/main/java/org/radarbase/output/source/SourceStorageFactory.kt

```diff
@@ -1,49 +1,35 @@
 package org.radarbase.output.source
 
 import com.azure.storage.blob.BlobServiceClient
-import io.minio.MinioClient
 import org.radarbase.output.config.ResourceConfig
 import org.radarbase.output.config.ResourceType
 import java.nio.file.Path
 
 class SourceStorageFactory(
-    private val resourceConfig: ResourceConfig,
     private val tempPath: Path,
 ) {
-    private val s3SourceClient: MinioClient? = if (resourceConfig.sourceType == ResourceType.S3) {
-        requireNotNull(resourceConfig.s3) { "Missing S3 configuration" }
-            .createS3Client()
-    } else {
-        null
-    }
-
-    private val azureSourceClient: BlobServiceClient? =
-        if (resourceConfig.sourceType == ResourceType.AZURE) {
-            requireNotNull(resourceConfig.azure) { "Missing Azure configuration" }
-                .createAzureClient()
-        } else {
-            null
-        }
+    fun createSourceStorage(consolidatedSources: List<ResourceConfig>) = consolidatedSources.map { createSourceStorage(it) }
 
-    fun createSourceStorage() = when (resourceConfig.sourceType) {
+    fun createSourceStorage(sourceConfig: ResourceConfig) = when (sourceConfig.sourceType) {
         ResourceType.S3 -> {
-            val s3Config = requireNotNull(resourceConfig.s3) { "Missing S3 configuration for source storage" }
-            val minioClient = requireNotNull(s3SourceClient) { "Missing S3 client configuration for source storage" }
-            S3SourceStorage(minioClient, s3Config, tempPath)
+            val s3Config =
+                requireNotNull(sourceConfig.s3) { "Missing S3 configuration for source storage" }
+            val minioClient = s3Config.createS3Client()
+            S3SourceStorage(sourceConfig.path, minioClient, s3Config, tempPath)
         }
         ResourceType.HDFS -> {
```

Comment:
Out of curiosity, we just removed HDFS from another repository, we do support it here?

Reply:
No I can remove it here as well, that will probably reduce merge conflicts.

Reply:
Merged it in.